### DISCUSSION TITLE
add command to ping blocks

### DISCUSSION THREAD

**Discussion by Huber1** [EssentialsX/Essentials#4974] [Ideas and Feature Suggestions] [answered]
Hello Everyone,
recently I stumbled across [this mod](https://www.curseforge.com/minecraft/mc-mods/ping) that enables you to ping blocks which are then highlighted to other Players. When playing with others it's an easy way to communicate locations.
I know that mods and plugins are quite different, but I think it could be possible to recreate something similar with a plugin.
This could be a command like `/pingblock` which spawns some particles for a few seconds at the location the player is currently looking at
I'm new to Spigot but I have some java knowledge so I'd like to try if I could come up with a possible working solution

**Comment by osfanbuff63** [comment]
I'd be happy with this! Feel free to work on that in a fork of this repo, but I still think it was the right thing to make this first

**Comment by mdcfe** [comment, marked as answer]
I'm not convinced this belongs in EssentialsX.

I can see some use cases for being able to ping locations in certain co-operative gameplay scenarios, but I don't think those limited cases warrant inclusion within EssentialsX. It also wouldn't be particularly intuitive without a client-side mod to render markers (particles are easily missed and won't render through solid blocks) or send pings on a keybind (pings are meant for immediate communications, but stopping to type a command loses that immediacy). A client mod is definitely out if scope for us at this time, and so we would be stuck with a very limited server-side version of location pings.

**Comment by Huber1** [comment]
Actually it works quite well
![image](https://user-images.githubusercontent.com/22856090/176873096-e5ab4557-9a82-44fc-bcb4-dc2c8833e8e4.png)

you are right about the command, another option could be double/triple-sneak to trigger the effect (with a config to disable the whole thing)
The problem that still persists is the render, the picture above shows a particle beam that is visible from far but not through blocks (doesn't work in caves then). Even if particles are set to minimal in render settings, it is shown
Originally I wanted to create a Beacon Fire but after reading some forums it seems pretty much impossible to create from the server.
If you had any ideas how it could be achieved though, I'd definitely try them out

**Reply by pop4959** [reply]
Looks cool, you should make it a plugin and post on Spigot :)

You won't be able to get beacons to render through solid blocks though as that is a change you would need to make on the client.

**Reply by Huber1** [reply]
so you think I should put it into a standalone plugin instead of adding it to Essentials?
I can do that, and if it should work well and get used I could merge it later then
